README.md

```diff
@@ -0,0 +1,122 @@
+for example:
+
+    aws-vault exec your-profile-name -- aws s3 ls
+
+### Set up your environment
+
+We store our Terraform state in an S3 bucket. Create
+and enable versioning on this bucket before you run any other commands.
+
+    export TERRAFORM_BUCKET=prometheus-staging
+
+    aws s3 mb "s3://${TERRAFORM_BUCKET}"
+
+    aws s3api put-bucket-versioning  \
+      --bucket ${TERRAFORM_BUCKET} \
+      --versioning-configuration Status=Enabled
+
+Now you have a bucketname you will create the configurarion for your
+stack. Inside the `environments` directory you will find a pair of files
+for each stack, a `.backend` and a `.tfvars`. Make a copy of an existing
+pair and change the values to suit your new name. The `bucket`
+and `remote_state_bucket` settings in these files must match the bucket you
+created above.
+
+### Creating your environment
+
+Once you've created your environment configurations, and added the
+correct bucket name, you can create the environment.
+
+Create the environment:
+
+    cd terraform/projects/infra-networking
+
+    $ terraform init -backend-config=../../../environments/staging.backend
+
+    $ terraform plan -var-file=../../../environments/staging.tfvars
+
+    $ terraform apply -var-file=../../../environments/staging.tfvars
+
+    cd ../infra-security-groups
+
+    # terraform commands from above
+
+    cd ../app-ecs-nodes
+
+    # terraform commands from above
+
+    cd ../app-ecs-albs
+
+    # terraform commands from above
+
+    cd ../app-ecs-services
+
+    # terraform commands from above
+
+## Creating documentation
+
+The projects in this repo use the [terraform-docs](https://github.com/segmentio/terraform-docs)
+to generate the per project documentation.
+
+You can install `terraform-docs` by running:
+
+    brew install terraform-docs
+
+When adding adding or changing terraform projects you should run
+
+    terraform-docs markdown . > README.md
+
+in the project directory and add that to your commit.
+
+## AWS Vault tips
+
+Every time you execute AWS vault it may ask for the credentials to access the keychain.
+
+To avoid this, we can follow the next steps (in OS X):
+
+1. Open the 'Keychain Access' utility.
+2. In the menu select "File > Add Keychain..."
+3. Select the "aws-vault.keychain". It can be found in `~/Library/Keychains/`.
+4. Once it is added, right click in it (it shows in the left hand side under 'Keychains') and select `Change Settings for Keychain aws-vault`.
+5. Uncheck `Lock after X minutes of inactivity` and `Look when sleeping`
+
+After this change, your credentials should be only asked the first time you use the tool after start/restart the machine.
+
+## ECS
+
+### Newest ECS AMI
+
+To see the latest ECS Optimized Amazon Linux AMI information in your
+default region, run this AWS CLI command:
+
+    aws ssm get-parameters --names /aws/service/ecs/optimized-ami/amazon-linux/recommended
```

Comment:
@deanwilson we copied this across from your spike repo, when is this supposed to be used?

Reply:
This would be used when moving to an updated ECS AMI, it stops you having to click through the AWS website docs for an hour.

https://github.com/deanwilson/ecs-monitoring/blob/master/terraform/projects/app-ecs-nodes/main.tf#L23

You can do some lookup magic to make EC2 pick the newest AMI but I intensely dislike that as you don't know what your base AMI is.

Reply:
Thanks, have added a small sentence to clarify when to use this.